drivers/net/mlx5/mlx5_vdpa.c

```diff
@@ -18,22 +18,30 @@
 /** Driver Static values in the absence of device VIRTIO emulation support */
 #define MLX5_VDPA_SW_MAX_VIRTQS_SUPPORTED 1
 
+#define MLX5_VDPA_FEATURES ((1ULL << VHOST_USER_F_PROTOCOL_FEATURES) | \
+			    (1ULL << VIRTIO_F_VERSION_1) | \
+			    (1ULL << VHOST_USER_PROTOCOL_F_SLAVE_REQ) | \
+			    (1ULL << VHOST_USER_PROTOCOL_F_SLAVE_SEND_FD) | \
+			    (1ULL << VHOST_USER_PROTOCOL_F_HOST_NOTIFIER));
```

Comment:
Those last 3 (VHOST_USER_PROTOCOL*) are VHOST protocol features and not VIRTIO features, so they need to go in a separate 64 bitmask, as we need to return them from get_protocol_features (and not from get_features).
The first one VHOST_USER_F_PROTOCOL_FEATURES is indeed in the VIRTIO features to negotiate if protocol features are supported at all.

Reply:
OK. will adjust.